### PR TITLE
Merge/3.x

### PR DESCRIPTION
**Pull request by jordisala1991**
I am targetting this branch, because I am doing manual merge with 3.x.

## Subject

Doing a manual merge since there are a lot of conflicts. Do I have to remove all the NEXT MAJOR tags here or we do it in another PR after this gets merged?

### CODE REVIEW

**Comment by greg0ire** [comment]
`NEXT_MAJOR` should be cleaned in another PR.

File: Admin/BaseMediaAdmin.php [hunk 1]
```diff
@@ -106,7 +106,17 @@ public function getNewInstance()
...
+
+                if (method_exists('Symfony\Component\HttpFoundation\JsonResponse', 'transformJsonError')) {
+                    // NEXT_MAJOR remove this block when dropping sf < 2.8 compatibility
+                    $media->setProviderName(
+                        $this->getRequest()->get(sprintf('%s[providerName]', $uniqid), null, true)
+                    );
+                } else {
+                    $providerParams = $this->getRequest()->get($uniqid);
+                    $media->setProviderName($providerParams['providerName']);
+                }
```
Comment (greg0ire) [inline Admin/BaseMediaAdmin.php]: Not sure you did the right thing here… If you got a merge conflict here, that might be b/c someone (probably me) cleaned this in a previous commit on master. Let me check.

Comment (greg0ire) [inline Admin/BaseMediaAdmin.php]: No in fact this one is ok 👍

Comment (jordisala1991) [inline Admin/BaseMediaAdmin.php]: $media->setProviderName($this->getRequest()->get('provider'));

in master this is the code... not sure, that wasn't the same as 3.x

**Comment by jordisala1991** [comment]
Added last PR pushed, but I think I fucked up with merge 😭

**Comment by greg0ire** [comment]
Then `reset --hard` on 35b18d32b69efa9c02b466b300968b5684c20d1f and start over :P

**Comment by jordisala1991** [comment]
Not sure how to add last PR without this commit:

`Merge remote-tracking branch 'upstream/3.x' into merge/3.x`

**Comment by greg0ire** [comment]
I think you are supposed to have it, it's the merge commit you are creating. If you really don't want it, try `merge --ff`

**Comment by jordisala1991** [comment]
Then its okay, I thought I did something wrong

**Comment by jordisala1991** [comment]
Added last two PR here

**Comment by greg0ire** [comment]
I'll merge when the build is green, we shouldn't let merges rot that long.

**Comment by jordisala1991** [comment]
Thanks, after this we can merge dev-kit PR also

**Comment by greg0ire** [comment]
Thanks @jordisala1991 !

**Pull request merged**